src/fhirtypes/ElementDefinition.ts

```diff
@@ -1642,8 +1643,16 @@ export class ElementDefinition {
    * @throws {InvalidUriError} when the system being assigned is not a valid uri
    */
   private assignFshCode(code: FshCode, exactly = false, fisher?: Fishable): void {
-    if (code.system && !isUri(code.system.split('|')[0])) {
-      throw new InvalidUriError(code.system);
+    if (code.system) {
+      const csURI = code.system.split('|')[0];
+      if (fisher) {
+        const vsURI = fisher.fishForMetadata(code.system, Type.ValueSet)?.url.split('|')[0] ?? '';
+        if (isUri(vsURI)) {
+          throw new MismatchedBindingTypeError(code.system, this.path, 'Code System');
```

Comment:
I addressed this in the most recent commit; though I opted to just change the language used in `MismatchedBindingTypeError` when the target type is a Code System rather than creating a new error class. If you still think that it's weird to throw a `MismatchedBindingTypeError` in this case at all then I can create a new class!